framework/doc/content/syntax/FunctorMaterials/index.md

```diff
@@ -0,0 +1,169 @@
+where here `_foo` is a functor material property data member of the kernel. The
+functor material property system introduces APIs very similar to the traditional
+material property system for declaring and getting properties. To declare a
+functor property:
+
+- `declareFunctorProperty<TYPE>`
+
+where `TYPE` can be anything such as `Real, ADReal, RealVectorValue, ADRealVectorValue`
+etc. To get a functor material property:
+
+- `getFunctor<TYPE>`
+
+It's worth noting that whereas the traditional regular material property system
+has different methods to declare/get non-AD and AD properties, the new functor
+system has single APIs for both non-AD and AD property types.
+
+Currently, functor material property evaluations are defined using the API:
+
+```c++
+template <typename T>
+template <typename PolymorphicLambda>
+void FunctorMaterialProperty<T>::
+setFunctor(const MooseMesh & mesh,
+           const std::set<SubdomainID> & block_ids,
+           PolymorphicLambda my_lammy);
+```
+
+where the first two arguments are used to setup block restriction and the last argument is a lambda
+defining the property evaluation. The lambda must be callable with two arguments, the first
+corresponding to space, and the second corresponding to time, and must return the type `T` of the
+`FunctorMaterialProperty`. An example of setting a constant functor material property that returns
+an `ADReal` looks like:
+
+```c++
+    _constant_unity_prop.setFunctor(
+        _mesh, blockIDs(), [](const auto &, const auto &) -> ADReal { return 1.; });
+```
+
+An example of a functor material property that depends on a nonlinear variable would look like
+
+```c++
+    _u_prop.setFunctor(_mesh, blockIDs(), [this](const auto & r, const auto & t) -> ADReal {
+      return _u_var(r, t);
+    });
+```
+
+In the above example, we simply forward the calling arguments along to the variable. Variable
+functor implementation is described in [MooseVariableBase.md#functor-vars]. A test functor material
+class to setup a dummy Euler problem is shown in
+
+!listing test/src/materials/ADCoupledVelocityMaterial.C
+
+In the following subsections, we describe the various spatial arguments that functor (material
+properties) can be evaluated at. Almost no functor material developers should have to concern
+themselves with these details as most material property functions should just appear as functions of
+space and time, e.g. the same lambda defining the property evaluation should apply across all
+spatial and temporal arguments. However, in the case that a functor material developer wishes to
+create specific implementations for specific arguments (as illustrated in `IMakeMyOwnFunctorProps`
+test class) or simply wishes to know more about the system, we give the details below.
+
+Any call to a functor (material property) looks like the following
+`_foo(const SpatialArg & r, const TemporalArg & t)`. Below are the possible type overloads of
+`SpatialArg`.
+
+### FaceArg id=spatial-overloads
```

Comment:
Maybe we should move all of this to the new Functors `index.md`. The following will also need to be updated

Reply:
And I'm happy to help with the updates

Reply:
Oh it looks like we already do have it there. So let's remove this duplicate stuff